joomla/joomla-cms
# [3.10] Compatibilitycheck

Pull Request for Issue # .

### Summary of Changes

The current upgrade compatibility checks for extensions is not easy to understand, the results are not grouped together in any meaningful way, they also do not provide a comparison between available compatible versions for currently installed version of Joomla and the targetted upgrade version.

### Testing Instructions

1. Apply patch to Joomla 3.10-dev branch
2. Set the upgrade server settings for Joomla to target one of the 4.x releases e.g. https://update.joomla.org/core/nightlies/next_major_list.xml
3. check for new Joomla upgrades

For convenience only I have provided a set of 'dummy' extensions to illustrate the different sort of results you could achieve - please just install them before checking for new Joomla upgrades.

[testuptodatepackage.zip](https://github.com/joomla/joomla-cms/files/4751464/testuptodatepackage.zip)
[testwuj3ej4package.zip](https://github.com/joomla/joomla-cms/files/4751465/testwuj3ej4package.zip)
[testuxpackage.zip](https://github.com/joomla/joomla-cms/files/4751466/testuxpackage.zip)
[testwuj3n4package.zip](https://github.com/joomla/joomla-cms/files/4751467/testwuj3n4package.zip)
[testwuvcpackage.zip](https://github.com/joomla/joomla-cms/files/4751468/testwuvcpackage.zip)
[testxupackage.zip](https://github.com/joomla/joomla-cms/files/4751469/testxupackage.zip)

The results will look like this screenshot.

![Screenshot_2020-06-09 Joomla Update - Joomla 3 10 Git - Administration](https://user-images.githubusercontent.com/839810/84135378-d0bf1000-aa41-11ea-996c-c37af6b1bdf9.png)

### Documentation Changes Required
Pre-update check docs will need amending
